### PR TITLE
Incremental backup GC error

### LINKED ISSUE #8994: backup: Add the gc / protected timestamp issues to the trouble shooting section
We have really bad error messages when your data has been GC-ed and therefore you can't take an incremental backup. We should doc that to make clear how to intepret and how a user should respont.

* Add the error message for searchability
* Add the triage steps

### PR DESCRIPTION
**Pull request by gemma-shay**
Add error to Common Errors page
Add links to relevant callouts

Closes #8994

### CODE REVIEW

**Review by lnhsingh — COMMENTED** [review]
LGTM, with a small nit that applies to all occurrences 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/9916#-:-MUsT-pg-QQ03Rgu8We3:b-ov77sv)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @gemma-shay)

___
*[v20.2/backup.md, line 104 at r2](https://reviewable.io/reviews/cockroachdb/docs/9916#-MUsSKHP1ajWD8G1SssB:-MUsSKHP1ajWD8G1SssC:bnx7deh) ([raw file](https://github.com/cockroachdb/docs/blob/b811596bdc9dd7540d6068ba7264e0d469f4b5f3/v20.2/backup.md#L104)):*
> ```Markdown
> 
> {{site.data.alerts.callout_danger}}
> Incremental backups can only be created within the garbage collection period of the base backup's most recent timestamp. This is because incremental backups are created by finding which data has been created or modified since the most recent timestamp in the base backup––that timestamp data, though, is deleted by the garbage collection process. Incremental backups that fail because of a recent garbage collection will throw [this error](common-errors.html#pq-failed-to-verify-protection-id).
> ```

nit: `this error` > `an error` (edit below too)

**Comment by lnhsingh** [comment]
@mwang1026 - mind doing a quick review on this?

**Comment by mwang1026** [comment]
Can we regenerate the previews? How does one do so? :D

**Review by mwang1026 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/9916)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @gemma-shay)

*[a discussion](https://reviewable.io/reviews/cockroachdb/docs/9916#-MWVIKZRAolHQjwMnbor:-MWVIKZRAolHQjwMnbos:bmmud3y) (no related file):*
A few notes there and clarifying how exactly to "grep" the table/range that's causing issues.

One thing to note is that we have a NEW error message in 21.1 that @adityamaru  just pushed out https://github.com/cockroachdb/cockroach/pull/62412. I haven't actually seen the new error in action so I can try to repro unless you have one handy Aditya :smile: 

---
___
*[v20.2/common-errors.md, line 146 at r3](https://reviewable.io/reviews/cockroachdb/docs/9916#-MWVGXhb0wxFPjh88hIF:-MWVGXhb0wxFPjh88hIG:b-wczsp) ([raw file](https://github.com/cockroachdb/docs/blob/b54b5e22345f8e496a1114e5844fc05c24158e51/v20.2/common-errors.md#L146)):*
> ```
> garbage collection period of the base backup's most recent timestamp
> ```

gc.ttlseconds is a zone configuration that can be set on anything -- a range, a table, a database. I think it's important to call out it might just be *one table* in there whose data has been gc-ed.

In the error message, it makes reference to `/Table/#` which points to which table is the problem. We probably should point them to also be able to inspect which tables/etc they hit the error on (you can take the  table ID and run `select * from crdb_internal.tables where id=<thetableID>` and get that.

You can also run `SHOW ZONE CONFIGURATIONS` to see which gc.ttlseconds is set quite low
___
*[v20.2/common-errors.md, line 146 at r3](https://reviewable.io/reviews/cockroachdb/docs/9916#-MWVI0_sB61bzf8FAOUB:-MWVI0_sB61bzf8FAOUC:b-l6y1dy) ([raw file](https://github.com/cockroachdb/docs/blob/b54b5e22345f8e496a1114e5844fc05c24158e51/v20.2/common-errors.md#L146)):*
> ```
> To resolve this issue
> ```

I think this is more to resolve this issue from recurring. There is a "one time fix" which is you will never run into this issue running FULL Backups, so if you had some onetime gc.ttlseconds blip or if you do end up increasing your gc.ttlseconds setting, your incremental backups will continue to fail until you take a new "base" full backup

**Review by mwang1026 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/9916)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @gemma-shay)

___
*[v20.2/common-errors.md, line 146 at r3](https://reviewable.io/reviews/cockroachdb/docs/9916#-MWVGXhb0wxFPjh88hIF:-MWVMV4u7dstlOj3hMYe:b-e6feak) ([raw file](https://github.com/cockroachdb/docs/blob/b54b5e22345f8e496a1114e5844fc05c24158e51/v20.2/common-errors.md#L146)):*
<details><summary><i>Previously, mwang1026 (Michael Wang) wrote…</i></summary><blockquote>

> ```
> garbage collection period of the base backup's most recent timestamp
> ```

gc.ttlseconds is a zone configuration that can be set on anything -- a range, a table, a database. I think it's important to call out it might just be *one table* in there whose data has been gc-ed.

In the error message, it makes reference to `/Table/#` which points to which table is the problem. We probably should point them to also be able to inspect which tables/etc they hit the error on (you can take the  table ID and run `select * from crdb_internal.tables where id=<thetableID>` and get that.

You can also run `SHOW ZONE CONFIGURATIONS` to see which gc.ttlseconds is set quite low
</blockquote></details>

To clarify this, `/Table/73{8-9}` indicates table `738-739`

**Review by gemma-shay — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/9916)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @gemma-shay and @mwang1026)

___
*[v20.2/common-errors.md, line 146 at r3](https://reviewable.io/reviews/cockroachdb/docs/9916#-MWVI0_sB61bzf8FAOUB:-MW_8HSd0FEvvVg_VHUF:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/b54b5e22345f8e496a1114e5844fc05c24158e51/v20.2/common-errors.md#L146)):*
<details><summary><i>Previously, mwang1026 (Michael Wang) wrote…</i></summary><blockquote>

> ```
> To resolve this issue
> ```

I think this is more to resolve this issue from recurring. There is a "one time fix" which is you will never run into this issue running FULL Backups, so if you had some onetime gc.ttlseconds blip or if you do end up increasing your gc.ttlseconds setting, your incremental backups will continue to fail until you take a new "base" full backup
</blockquote></details>

Done.
___
*[v20.2/common-errors.md, line 146 at r3](https://reviewable.io/reviews/cockroachdb/docs/9916#-MWVGXhb0wxFPjh88hIF:-MW_8ICi5kgCF9DPmdCl:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/b54b5e22345f8e496a1114e5844fc05c24158e51/v20.2/common-errors.md#L146)):*
<details><summary><i>Previously, mwang1026 (Michael Wang) wrote…</i></summary><blockquote>

To clarify this, `/Table/73{8-9}` indicates table `738-739`
</blockquote></details>

Done.

**Review by mwang1026 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/9916)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @gemma-shay and @mwang1026)

___
*[v20.2/common-errors.md, line 148 at r4](https://reviewable.io/reviews/cockroachdb/docs/9916#-MWdthQQ1_FdvqfD1oQZ:-MWdthQQ1_FdvqfD1oQ_:b-raz6pf) ([raw file](https://github.com/cockroachdb/docs/blob/e44d42d84e7860af77db1a5fdca468be757b5ac7/v20.2/common-errors.md#L148)):*
> ```
> You can then inspect this table by running [`SELECT * FROM example_db.tables WHERE id=771`]
> ```

I think this is specifically to look up the table name (and I think it has to be crdb_internal.tables) vs. example_db.tables. I think :D
___
*[v20.2/common-errors.md, line 150 at r4](https://reviewable.io/reviews/cockroachdb/docs/9916#-MWdzKAm1iuIRYa2et9X:-MWdzKAm1iuIRYa2et9Y:bpauu0b) ([raw file](https://github.com/cockroachdb/docs/blob/e44d42d84e7860af77db1a5fdca468be757b5ac7/v20.2/common-errors.md#L150)):*
> ```
> To resolve this issue, you can take a new [full backup](take-full-and-incremental-backups.html) before retrying your incremental backup.
> ```

I would rephrase this a bit. You really need to do BOTH to resolve the issue. Maybe something like "incremental backups will continue to fail until you (1) either increase GC or increase backup frequency [note INCREASE frequency, not decrease] and (2) take a new full backup and restart backup chain"
___
*[v21.1/common-errors.md, line 146 at r4](https://reviewable.io/reviews/cockroachdb/docs/9916#-MWdzkqY6uAy0HyN3dTJ:-MWdzkqY6uAy0HyN3dTK:bsp2nbt) ([raw file](https://github.com/cockroachdb/docs/blob/e44d42d84e7860af77db1a5fdca468be757b5ac7/v21.1/common-errors.md#L146)):*
> ```
> `pq: failed to verify protection id…`
> ```

It's a different error now in 21.1. Let me try to see if I can fish up an error.

**Review by gemma-shay — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/9916)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @gemma-shay and @mwang1026)

___
*[v21.1/common-errors.md, line 146 at r4](https://reviewable.io/reviews/cockroachdb/docs/9916#-MWdzkqY6uAy0HyN3dTJ:-MWe6qcu82YfnSxn95cs:bxioxhu) ([raw file](https://github.com/cockroachdb/docs/blob/e44d42d84e7860af77db1a5fdca468be757b5ac7/v21.1/common-errors.md#L146)):*
<details><summary><i>Previously, mwang1026 (Michael Wang) wrote…</i></summary><blockquote>

> ```
> `pq: failed to verify protection id…`
> ```

It's a different error now in 21.1. Let me try to see if I can fish up an error.
</blockquote></details>

just got the new error from Aditya, will update in a minute!

**Review by gemma-shay — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/9916)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @gemma-shay and @mwang1026)

___
*[v20.2/common-errors.md, line 148 at r4](https://reviewable.io/reviews/cockroachdb/docs/9916#-MWdthQQ1_FdvqfD1oQZ:-MWeloLKA334fZZlxJ1T:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/e44d42d84e7860af77db1a5fdca468be757b5ac7/v20.2/common-errors.md#L148)):*
<details><summary><i>Previously, mwang1026 (Michael Wang) wrote…</i></summary><blockquote>

> ```
> You can then inspect this table by running [`SELECT * FROM example_db.tables WHERE id=771`]
> ```

I think this is specifically to look up the table name (and I think it has to be crdb_internal.tables) vs. example_db.tables. I think :D
</blockquote></details>

Done.
___
*[v20.2/common-errors.md, line 150 at r4](https://reviewable.io/reviews/cockroachdb/docs/9916#-MWdzKAm1iuIRYa2et9X:-MWelsQ2C_7CKbbFObSy:b-fm8ctf) ([raw file](https://github.com/cockroachdb/docs/blob/e44d42d84e7860af77db1a5fdca468be757b5ac7/v20.2/common-errors.md#L150)):*
<details><summary><i>Previously, mwang1026 (Michael Wang) wrote…</i></summary><blockquote>

> ```
> To resolve this issue, you can take a new [full backup](take-full-and-incremental-backups.html) before retrying your incremental backup.
> ```

I would rephrase this a bit. You really need to do BOTH to resolve the issue. Maybe something like "incremental backups will continue to fail until you (1) either increase GC or increase backup frequency [note INCREASE frequency, not decrease] and (2) take a new full backup and restart backup chain"
</blockquote></details>

How's this update?
___
*[v21.1/common-errors.md, line 146 at r4](https://reviewable.io/reviews/cockroachdb/docs/9916#-MWdzkqY6uAy0HyN3dTJ:-MWemG2_A1XidVX6pKrD:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/e44d42d84e7860af77db1a5fdca468be757b5ac7/v21.1/common-errors.md#L146)):*
<details><summary><i>Previously, gemma-shay (Gemma Shay) wrote…</i></summary><blockquote>

just got the new error from Aditya, will update in a minute!
</blockquote></details>

Done.

**Review by lnhsingh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/9916)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @adityamaru and @mwang1026)

*[a discussion](https://reviewable.io/reviews/cockroachdb/docs/9916#-MWVIKZRAolHQjwMnbor:-MWjuLfP7DBA6TqIKZ7L:b-fiy556) (no related file):*
<details><summary><i>Previously, mwang1026 (Michael Wang) wrote…</i></summary><blockquote>

A few notes there and clarifying how exactly to "grep" the table/range that's causing issues.

One thing to note is that we have a NEW error message in 21.1 that @adityamaru  just pushed out https://github.com/cockroachdb/cockroach/pull/62412. I haven't actually seen the new error in action so I can try to repro unless you have one handy Aditya :smile: 
</blockquote></details>

One suggestion on wording but otherwise LGTM!

---
___
*[v20.2/common-errors.md, line 150 at r4](https://reviewable.io/reviews/cockroachdb/docs/9916#-MWdzKAm1iuIRYa2et9X:-MWjtTnZ9hUHbLL-XUQ0:bjol7ea) ([raw file](https://github.com/cockroachdb/docs/blob/e44d42d84e7860af77db1a5fdca468be757b5ac7/v20.2/common-errors.md#L150)):*
<details><summary><i>Previously, gemma-shay (Gemma Shay) wrote…</i></summary><blockquote>

How's this update?
</blockquote></details>

To make this section a little clearer, I suggest changing the wording around a little:

```
To resolve this issue, take a new full backup, then do one of the following:

- Increase the gc period...
- Decrease the frequency...
```

**Review by gemma-shay — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/9916)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @adityamaru and @mwang1026)

___
*[v20.2/common-errors.md, line 150 at r4](https://reviewable.io/reviews/cockroachdb/docs/9916#-MWdzKAm1iuIRYa2et9X:-MWk1v0k6iAbTAj1BIav:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/e44d42d84e7860af77db1a5fdca468be757b5ac7/v20.2/common-errors.md#L150)):*
<details><summary><i>Previously, lnhsingh (Lauren Hirata Singh) wrote…</i></summary><blockquote>

To make this section a little clearer, I suggest changing the wording around a little:

```
To resolve this issue, take a new full backup, then do one of the following:

- Increase the gc period...
- Decrease the frequency...
```

</blockquote></details>

Done.

**Review by mwang1026 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/9916)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @adityamaru, @gemma-shay, and @mwang1026)

___
*[v20.2/common-errors.md, line 150 at r6](https://reviewable.io/reviews/cockroachdb/docs/9916#-MXEPjPaDipUk5c3McjP:-MXEPjPaDipUk5c3McjQ:bq791vk) ([raw file](https://github.com/cockroachdb/docs/blob/077f9c07fdd9bd16561a40746e3e07d40446e9a2/v20.2/common-errors.md#L150)):*
> ```Markdown
> The error message will specify which part of your backup is causing the failure. For example, `r13485:/Table/771` indicates that table `771` is part of the problem. You can then inspect this table by running [`SELECT * FROM crdb_internal.tables WHERE id=771`](select-clause.html). You can also run [`SHOW ZONE CONFIGURATIONS`](show-zone-configurations.html) and look for any `gc.ttlseconds` values that are set lower than your incremental backup frequency.
> 
> To resolve this issue, take a new [full backup](take-full-and-incremental-backups.html), then do either of the following:
> ```

Technically, it's safer to do the below things FIRST, then take a new full to reset the backup chain.
___
*[v20.2/common-errors.md, line 153 at r6](https://reviewable.io/reviews/cockroachdb/docs/9916#-MXEPYyH9yy-d9WIkU47:-MXEPYyH9yy-d9WIkU48:bjvtope) ([raw file](https://github.com/cockroachdb/docs/blob/077f9c07fdd9bd16561a40746e3e07d40446e9a2/v20.2/common-errors.md#L153)):*
> ```Markdown
> 
> - Increase the garbage collection period by [configuring the `gc.ttlseconds` replication zone variable](configure-replication-zones.html#gc-ttlseconds), or
> - [Decrease the frequency of incremental backups](manage-a-backup-schedule.html).
> ```

This should either be "Decrease the time between incremental backups" or "Increase the incremental backup frequency". Right now it's the wrong advice :joy:

**Review by gemma-shay — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/9916)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @adityamaru and @mwang1026)

___
*[v20.2/common-errors.md, line 150 at r6](https://reviewable.io/reviews/cockroachdb/docs/9916#-MXEPjPaDipUk5c3McjP:-MXcRhVcEGkpGgT-yn4Y:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/077f9c07fdd9bd16561a40746e3e07d40446e9a2/v20.2/common-errors.md#L150)):*
<details><summary><i>Previously, mwang1026 (Michael Wang) wrote…</i></summary><blockquote>

Technically, it's safer to do the below things FIRST, then take a new full to reset the backup chain.
</blockquote></details>

Done.
___
*[v20.2/common-errors.md, line 153 at r6](https://reviewable.io/reviews/cockroachdb/docs/9916#-MXEPYyH9yy-d9WIkU47:-MXcSAQCB0mMntrGf6Cr:b-sjblhw) ([raw file](https://github.com/cockroachdb/docs/blob/077f9c07fdd9bd16561a40746e3e07d40446e9a2/v20.2/common-errors.md#L153)):*
<details><summary><i>Previously, mwang1026 (Michael Wang) wrote…</i></summary><blockquote>

This should either be "Decrease the time between incremental backups" or "Increase the incremental backup frequency". Right now it's the wrong advice :joy: 
</blockquote></details>

ah good catch!!
